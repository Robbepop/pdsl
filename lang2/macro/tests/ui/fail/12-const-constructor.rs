#![feature(proc_macro_hygiene)]

use ink_lang2 as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        const fn invalid_return(&mut self) {}

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
