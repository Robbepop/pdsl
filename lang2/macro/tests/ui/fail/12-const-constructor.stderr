error: `const fn` is not supported for ink! functions
  --> $DIR/12-const-constructor.rs:12:9
   |
12 |         const fn invalid_return(&mut self) {}
   |         ^^^^^
